cromshell

```diff
@@ -402,12 +405,178 @@ function populateWorkflowIdAndServerUrl()
   fi
 }
 
+function _turtleSpinner()
+{
+  # Require a flag file to be given to this function.
+  # This keeps it safe.
+  if [[ $# -ne 1 ]] ; then
+    return 1
+  fi
+
+  local flagFile=$1
+
+  # Let's be fancy and make turtles that are walking while we wait:
+  local tmpTurtFile=$( makeTemp )
+  local turtFile=$( makeTemp )
+  local turtRFile=$( makeTemp )
+  turtle &> ${tmpTurtFile}
+  sed 's#^  ##g' ${tmpTurtFile} > ${turtFile}
+  rev ${turtFile} | sed -e 's$/$#$g' -e 's#\\#/#g' -e 's$#$\\$g' -e 's#(#)#g' -e 's#)#(#g' > ${turtRFile}
+
+  # Set up some loop variables:
+  local i=3
+  local incrementString="+1"
+  local curTurtFile=${turtFile}
+
+  # Let's loop while the ${flagFile} exists.
+  # Note that this will be spun off into its own thread:
+  while [ -f ${flagFile} ] ; do
+    # Create padding for turtle:
+    padding=$( printf "%${i}s" "" )
+
+    # Display our turtle, moved a little to the right:
+    cat ${curTurtFile} | sed -e "s#^#${padding}#g"
+    sleep 0.1
+
+    # Move the cursor back up to the top line of the turtle
+    # so we create the illusion of movement:
+    tput cuu 6
+
+    # Now we update our loop variables in a clever way
+    # so that the turtle will go back and forth.
+    let i=i${incrementString}
+    if [ $i -eq 15 ] ; then 
+      incrementString="-1" 
+      curTurtFile=${turtRFile}
+    elif [ $i -eq 0 ] ; then
+      incrementString="+1" 
+      curTurtFile=${turtFile}
+    fi
+
+  done & # The & here spins this off as a background thread/process
+}
+
+function waitOnSubmittedStatus() 
+{
+  local id=$1
+  local cromwellServerUrl=$2
+
+  echo "Waiting on status to progress from 'Submitted' ..."
+  echo
+
+  # Use tput to save the cursor position:
+  tput sc
+
+  # The turtle starts 9 rows above where the cursor is at this point,
+  # so if we need to move it, we have to move the cursor there before
+  # we output anything:
+  tput cuu 9
```

Comment:
It's correct that this heigh is different than the wait below?

Reply:
Yes - the height is determined by the number of output lines between just before the turtle status is initially displayed and where we are now.  We output an extra line after the initial turtle status that we do not add below (L464: `echo "Waiting on status to progress from 'Submitted' ..."`).

When we return to the main process, the cursor is at the top of the turtle.  Since I want to overwrite the '{"id" ...}' line and the "Waiting on status" line.

{"id":"dadfc5e7-f366-4859-830d-dc2f5effcb1b","status":"Submitted"}
Waiting on status to progress from 'Submitted' ...